### ISSUE TITLE
Check if metrics are available for corresponding alert rules

### ISSUE THREAD

**Issue by DnPlas** [canonical/cos-tool#16]
### Enhancement Proposal

It would be interesting if this tool could verify if a charm with a set of alert rules is actually exposing the metrics those alert rules refer to.
This can be specially useful in the following cases:

* If the metric name changes
* If the metric is removed entirely, but the charm author did not remove the alert rule
* If the metric even exists

**Comment by sed-i** [comment]
Agreed. cos-tool would then need to be able to extract metric names from alert rules.

Just realized there could be some minor edge cases that may affect testing:
- When it comes to labels, some instrumentations have conditional metrics, e.g. api latency per path, and if the particular api endpoint wasn't probed yet, then the metric with the corresponding "path" label wouldn't exist.
- Also related are aggregation operators: it would be difficult to make test assertion that the labels used by aggregation operators actually exist.

**Comment by lucabello** [comment]
@DnPlas the edge cases that Leon mentioned are widespread enough to make this problematic, I think.

Do you have any idea on how we could approach this? Is this issue still needed/relevant?

**Comment by lucabello** [comment]
Closing, but please feel free to reopen if needed :)
